src/OrchardCore.Modules/OrchardCore.Facebook/Drivers/FacebookCoreSettingsDisplayDriver.cs

```diff
@@ -58,39 +54,28 @@ public override async Task<IDisplayResult> EditAsync(FacebookCoreSettings settin
 
             return Initialize<FacebookCoreSettingsViewModel>("FacebookCoreSettings_Edit", model =>
             {
+                var protector = _dataProtectionProvider.CreateProtector(FacebookConstants.Features.Core);
+
                 model.AppId = settings.AppId;
-                model.AppSecret = settings.AppSecret;
+                model.AppSecret = protector.Unprotect(settings.AppSecret);
 
             }).Location("Content:0").OnGroup(FacebookConstants.Features.Core);
         }
 
         public override async Task<IDisplayResult> UpdateAsync(FacebookCoreSettings settings, BuildEditorContext context)
         {
-            var user = _httpContextAccessor.HttpContext?.User;
```

Comment:
Should the other display driver be updated to perform the authorization check at the same stage?

Reply:
I looked for other patterns, but could only find the same one. I assume you are talking about having the security check inside the shape lambda.

Reply:
No, I meant why moving the authorization check inside the `if` block? It's not consistent with what exists elsewhere (but maybe it's the right thing to do, I dunno): 

https://github.com/OrchardCMS/OrchardCore/blob/80670ba7c474f0c0bae30b22bfbdd3f9bc82e95e/src/OrchardCore.Modules/OrchardCore.Email/Drivers/SmtpSettingsDisplayDriver.cs#L72-L82